### PR TITLE
Got TrajoptPlanner in working order

### PR DESCRIPTION
**Pull request by mpowelson**
I ran into the CMakeLists problem, but Reid had already fixed it here along with some other things.

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@mpowelson could you rebase this.

**Comment by mpowelson** [comment]
@Levi-Armstrong Just rebased it. Looks like Jonathan fixed the CMakeLists recently.

**Comment by mpowelson** [comment]
@Levi-Armstrong Just added the ability to add callbacks and change the parameters.

**Comment by mpowelson** [comment]
@Levi-Armstrong That should be all of the requested changes.

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 1]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
     // Converge Status Codes
   }
 
+  /**
+   * @brief solve Sets up the opimizer and solves a SQP problem read from json with no callbacks and dafault parameterss
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: remove 'solve' it is the default

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 2]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+   * @param response The results of the optimization. Primary output is the optimized joint trajectory
+   * @return true if optimization complete
+   */
   bool solve(PlannerResponse& response) override;
 
-  bool solve(trajopt::TrajOptProbPtr prob, PlannerResponse& response);
+  /**
+   * @brief solve Sets up the optimizer and solves the SQP problem with no callbacks and default parameters
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 3]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+   * @param response The results of the optimization. Primary output is the optimized joint trajectory
+   * @param prob Trajopt problem to be solved
+   * @return true if optimization complete
+   */
+  bool solve(PlannerResponse& response, const trajopt::TrajOptProbPtr prob);
 
-  bool solve(trajopt::TrajOptProbPtr prob, PlannerResponse& response, trajopt::BasicTrustRegionSQPParameters& params);
+  /**
+   * @brief solve Sets up the optimizer and solves the SQP problem with no callbacks and parameters passed in
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 4]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+  bool solve(PlannerResponse& response,
+             const trajopt::TrajOptProbPtr prob,
+             const trajopt::BasicTrustRegionSQPParameters& params);
 
-  bool solve(trajopt::TrajOptProbPtr prob,
-             PlannerResponse& response,
-             trajopt::BasicTrustRegionSQPParameters& params,
-             trajopt::Optimizer::Callback& callback);
+  /**
+   * @brief solve Sets up the optimizer and solves the SQP problem with a single callback and parameters passed in
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 5]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+   * @return true if optimization complete
+   */
+  bool solve(PlannerResponse& response,
+             const trajopt::TrajOptProbPtr prob,
+             const trajopt::BasicTrustRegionSQPParameters& params,
+             const trajopt::Optimizer::Callback& callback);
 
-  bool solve(trajopt::TrajOptProbPtr prob, PlannerResponse& response, trajopt::Optimizer::Callback& callback);
+  /**
+   * @brief solve Sets up the optimizer and solves the SQP problem with a single callback and default parameters
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 6]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+   */
+  bool solve(PlannerResponse& response,
+             const trajopt::TrajOptProbPtr prob,
+             const trajopt::Optimizer::Callback& callback);
 
-  bool solve(trajopt::TrajOptProbPtr prob,
-             PlannerResponse& response,
-             std::vector<trajopt::Optimizer::Callback>& callbacks);
+  /**
+   * @brief solve Sets up the optimizer and solves the SQP problem with a vector of callbacks and default parameters
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 7]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+             const std::vector<trajopt::Optimizer::Callback>& callbacks);
 
   /**
-   * @brief solve - Sets up optimizer and solves a SQP problem
-   * @param prob - trajopt problem to be solved
-   * @param response - the results of the optimization. Primary output is the optimized joint trajectory
-   * @param params - SQP parameters for the optimization. If not given, defaults will be used
-   * @param callbacks - Callbacks to be called on each iteration, e.g. plotting, write to file, etc.
-   * @return
+   * @brief solve Sets up optimizer and solves a SQP problem
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: same as above

File: tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h [hunk 8]
```diff
@@ -50,35 +50,83 @@ class TrajoptPlanner : public BasicPlanner
...
+   * @param callbacks A callback function to be called on each iteration of the optimization, e.g. plotting, write to
+   * file, etc.
+   * @return true if optimization complete
    */
-  bool solve(trajopt::TrajOptProbPtr prob,
-             PlannerResponse& response,
-             trajopt::BasicTrustRegionSQPParameters& params,
-             std::vector<trajopt::Optimizer::Callback>& callbacks);
+  bool solve(PlannerResponse& response,
+             const trajopt::TrajOptProbPtr prob,
```
Comment (Levi-Armstrong) [inline tesseract_planning/include/tesseract_planning/trajopt/trajopt_planner.h]: should be `const trajopt::TrajOptProbPtr& prob`

**Pull request merged**